src/Workspaces/VisualBasic/Portable/Extensions/SyntaxTreeExtensions.vb

```diff
@@ -114,6 +114,12 @@ recurse:
                 Return True
             End If
 
+            ' If it is not a char or string literal, we're okay
+            If Not token.IsKind(SyntaxKind.CharacterLiteralToken, SyntaxKind.StringLiteralToken) Then
```

Comment:
I'd much rather you do the reverse check.  i.e. "if isKind(integer, decimal, date, floating) return"